flink-state-backends/flink-statebackend-forst/src/main/java/org/apache/flink/state/forst/ForStKeyedStateBackend.java

```diff
@@ -291,7 +289,7 @@ public <N, S extends InternalKeyedState, SV> S createStateInternal(
                         new ForStValueState<>(
                                 stateRequestHandler,
                                 columnFamilyHandle,
-                                (ValueStateDescriptor<SV>) stateDesc,
+                                registerResult.f1.getStateSerializer(),
```

Comment:
Could you help update `namespaceSerializer` in `#updateRestoredStateMetaInfo`?

Reply:
👍Okay, let's do it in [FLINK-37553](https://issues.apache.org/jira/browse/FLINK-37553).